log-appender/src/main/java/io/phdata/pulse/log/HttpAppender.java

```diff
@@ -62,30 +68,37 @@ public void run() {
     } catch (UnknownHostException e) {
       LogLog.error("Could not set hostname: " + e, e);
     }
+
+    // Flush the buffer every second
+    int numCores = 1;
+    int initialDelay = 1;
+    final ScheduledExecutorService scheduler = Executors.newScheduledThreadPool(numCores);
+    scheduledFlushTask = scheduler.scheduleAtFixedRate(new FlusherTask(), initialDelay, FLUSH_PERIOD_SECONDS, TimeUnit.SECONDS);
+
   }
 
   @Override
   protected void append(LoggingEvent event) {
     try {
       bufferingEventHandler.addEvent(event);
-
       if (event.getProperty(HOSTNAME) == null) {
         event.setProperty(HOSTNAME, hostname);
       }
-
-      if (shouldFlush(event)) {
+      if (event.getLevel().isGreaterOrEqual(Level.ERROR)) {
         flush();
       }
+
     } catch (Throwable t) {
       LogLog.error("Unexpected error: " + t, t);
     }
   }
 
   /**
    * Flush all log events
+   *
    * @throws Exception
    */
-  private void flush() throws Exception {
+  void flush() throws Exception {
```

Comment:
This method is now called in parallel. Some of these variables are not protected. Is that ok?

Reply:
Maybe let the FlusherTask take care of calling this method, and remove the call from '.append()' to immediately flush?  Dunno, just an idea.